lib/Index/Index.cpp

```diff
@@ -825,10 +825,10 @@ static bool isTestCandidate(ValueDecl *D) {
 
   // 5. ...is of at least 'internal' accessibility (unless we can use
   //    Objective-C reflection)...
-#if SWIFT_OBJC_INTEROP
-  if (D->getFormalAccess() < Accessibility::Internal)
+  if (!D->getASTContext().LangOpts.EnableObjCInterop &&
+      (D->getFormalAccess() < Accessibility::Internal ||
+      parentNTD->getFormalAccess() < Accessibility::Internal))
```

Comment:
Yes, it is. @modocache's original work in #2518 included a test for this (https://github.com/apple/swift/blob/master/test/SourceKit/Indexing/index_is_test_candidate.swift#L17), although the implementation didn't have this required change until now. (I'm assuming that the original test was never actually executed because the pieces required to run this test on Linux are only now starting to come together.)